src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.cs

```diff
@@ -234,23 +236,26 @@ void ProcessTransaction(Transaction currentTx, int index)
             if (transactionsChangeable)
             {
                 int i = 0;
-                LinkedHashSet<Transaction> transactionsForBlock = new();
+                LinkedHashSet<Transaction> transactionsForBlock = new(ByHashTxComparer.Instance);
                 foreach (Transaction currentTx in transactions)
                 {
-                    TxAction txAction = CheckTx(currentTx, block).Action;
-
-                    if (txAction == TxAction.Skip)
-                    {
-                        continue;
-                    }
-                    
-                    if (txAction == TxAction.Stop)
+                    if (!transactionsForBlock.Contains(currentTx))
```

Comment:
Do we have test cases for this change?

Reply:
Yes - indirectly when producinb block with same transaction in different bundles

Reply:
Also its very intuitive, that we can't duplicate transactions in block